common/src/main/java/io/pravega/common/util/btree/ByteArrayComparator.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.util.btree;
+
+import io.pravega.common.util.ByteArraySegment;
+import java.io.Serializable;
+import java.util.Comparator;
+
+/**
+ * Compares two byte arrays of the same length.
+ */
+class ByteArrayComparator implements Comparator<byte[]>, Serializable {
```

Comment:
Why is this serializeable?

Reply:
Because FindBugs told me so. Apparently if you implement `Comparator` you also need to implement `Serializable`.